libs/prelude/Decidable/Equality.idr

```diff
@@ -30,6 +22,20 @@ interface DecEq t where
   total decEq : (x1 : t) -> (x2 : t) -> Dec (x1 = x2)
 
 --------------------------------------------------------------------------------
+-- Utility lemmas
+--------------------------------------------------------------------------------
+
+||| The negation of equality is symmetric (follows from symmetry of equality)
+total negEqSym : {a : t} -> {b : t} -> (a = b -> Void) -> (b = a -> Void)
+negEqSym p h = p (sym h)
+
+||| Everything is decidably equal to itself
+total decEqSelfIsYes : DecEq a => {x : a} -> decEq x x = Yes Refl
+decEqSelfIsYes {x} with (decEq x x)
```

Comment:
I'm doing an Idris translation of "Software foundations" and this has already come up twice when trying to prove things about functions defined via `decEq`. Typically you get to the point where the arguments are the same and then there's an error that smth like `Decidable.Equality.Nat implementation of Decidable.Equality.DecEq, method decEq n n` could not be reduced to get to the goal. Doing `rewrite decEqSelfIsYes {x=n}` solves this.